codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -184,4 +195,17 @@ public static String decodeName(ByteBuf in) {
 
         return name.toString();
     }
+
+    public static String decodeCharString(ByteBuf in) {
+        if (in.readableBytes() == 0) {
+            return null;
+        }
+        int length = (int) in.readByte();
+        if (in.readableBytes() < length) {
+            return null;
+        }
+        byte[] buf = new byte[length];
+        in.readBytes(buf);
+        return new String(buf, CharsetUtil.US_ASCII);
```

Comment:
Sure. I missed that one. It should be called readString(), btw, and advance readerIndex.